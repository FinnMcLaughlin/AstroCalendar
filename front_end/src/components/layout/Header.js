import React from 'react';
import { Link } from 'react-router-dom';

function Header(){
        return(
            <header data-testid="header-component" style={headerStyle}>
                <h1>Astro Calendar</h1>
                <Link style={linkStyle} to="/">Home</Link> | <Link style={linkStyle} to="/About">About</Link>
            </header>
        );
}

const headerStyle = {
    backgroundColor: '#090428',
    color: '#DFF649',
    paddingBottom: 10
}

const linkStyle = {
    color: '#DFF649'
}

export default Header;