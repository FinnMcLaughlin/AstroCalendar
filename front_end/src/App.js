import { BrowserRouter as Router, Route } from 'react-router-dom'
import React, { Component } from 'react'
import Header from './components/layout/Header.js'
import Events from './components/Events.js'
import About from './components/pages/About.js'
import SendEventReq from './services/SendEventReq.js'
import CreateEventObject from './services/CreateEventObj.js'
import ScrapeEvents from './services/ScrapeEvents.js'
import './App.css'

const LOCALHOST = 'http://localhost:5000'

const endpoints = {
  scrape_events: "/events",
  event_req: "/reminder"
}

class App extends Component {
  state = {
    events: []
  }

  componentDidMount(){
    console.log("Component Mounted")
    //this.scrapeEvents()
  }

  // Function to request the backend server to scrape the events from the given url 
  scrapeEvents = () => {
    ScrapeEvents(LOCALHOST + endpoints.scrape_events).then((events_data) => {
      console.log(events_data)
      this.setState({events: events_data})
    })
  }

  //Create calendar event reminder for specified event
  setReminder = (id) => {
    this.state.events.forEach((event) => {

      if(event.id === id){
        
        var event_object = CreateEventObject(event)

        if(!Array.isArray(event_object)){
          event_object = [event_object]
        }

        event_object.forEach(event => {
          SendEventReq(LOCALHOST + endpoints.event_req, event).then((resp) => {
            console.log(resp)
          })
        });
      }
    })

  }

  render(){
    return (
      <Router>
        <div className="App">
          <div className="container">
            
            {/* Home Page */}
            <Route exact path="/" render={props => (
              <React.Fragment>
                <Header />
                {/* Temporary Web Scrape Button */}
                <button onClick={this.scrapeEvents}>Get Events</button>
                <Events event_list={this.state.events} setReminder={this.setReminder}/> 
              </React.Fragment>
            )}/>

            {/* About Page */}
            <Route path="/About" render={() => (
              <React.Fragment>
              <Header />
              <About /> 
            </React.Fragment>
            )}/>

          </div>
        </div>
      </Router>
    );
  }
}

export default App;
