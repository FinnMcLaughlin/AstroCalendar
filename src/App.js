import { BrowserRouter as Router, Route } from 'react-router-dom'
import React, { Component } from 'react'
import Header from './components/layout/Header.js'
import Events from './components/Events.js'
import About from './components/pages/About.js'
import './App.css';

class App extends Component {
  state = {
    events: [
      {
        id: 1,
        event_name: 'Lunar Eclipse',
        event_date: '05/04/2021',
        reminder_set: false
      },
      {
        id: 2,
        event_name: 'Mars Visible',
        event_date: '18/04/2021',
        reminder_set: false
      },
      {
        id: 3,
        event_name: 'Meteor Shower',
        event_date: '11/06/2021',
        reminder_set: false
      }
    ]
  }

  //Toggle set reminder on event
  setReminder = (id) => {
    this.setState({
      events: this.state.events.map(event => {  
        
        if(event.id === id){
          event.reminder_set = event.reminder_set ? false : true; 
        }
        return event;  
      })
    })
  }

  render(){
    return (
      <Router>
        <div className="App">
          <div className="container">
            
            {/* Home Page */}
            <Route exact path="/" render={props => (
              <React.Fragment>
                <Header />
                <Events event_list={this.state.events} setReminder={this.setReminder}/> 
              </React.Fragment>
            )}/>

            {/* About Page */}
            <Route path="/About" render={props => (
              <React.Fragment>
              <Header />
              <About /> 
            </React.Fragment>
            )}/>

          </div>
        </div>
      </Router>
    );
  }
}

export default App;
