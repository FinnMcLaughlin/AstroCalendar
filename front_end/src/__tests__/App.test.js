import React from "react";
import { render, cleanup, fireEvent, waitFor} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import axiosMock from "axios";
import Header from '../components/layout/Header.js'
import App from '../App.js'

afterEach(() => {
    cleanup()
})

describe("App Component", () => {
    
    test("renders header correcty", () => {
        const { getByTestId } = render(<BrowserRouter><Header /></BrowserRouter>)
    
        const header_obj = getByTestId("header-component")
    
        expect(header_obj).toMatchSnapshot()
    })
    
    test("renders 'About' page correctly", () => {
        const { getByTestId } = render(<App page="/About" />)
    
        const app_object = getByTestId("app-component")
    
        expect(app_object).toMatchSnapshot()
    })
    
    test("renders 'Home' page correctly", async () => {
        axiosMock.get.mockResolvedValueOnce({ data:
            [{      
                id: 1,
                event_name: "Cosmic Insanity",
                event_details: "Hell of a trip man",
                event_date: "April 1",
                reminder_set: true
            },
            {      
                id: 2,
                event_name: "Don't Look Up",
                event_details: "Please, don't do it",
                event_date: "April 2",
                reminder_set: false
            }]
        });
    
        const { getByTestId } = render(<App />)
    
        fireEvent.click(getByTestId('scrape-button'))
    
        const app_component_obj = await waitFor(() => getByTestId("app-component"))
    
        expect(app_component_obj).toMatchSnapshot()
        expect(axiosMock.get).toHaveBeenCalledTimes(1);
    })
})

