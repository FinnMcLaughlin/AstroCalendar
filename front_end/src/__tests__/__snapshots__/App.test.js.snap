// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders 'About' page correctly 1`] = `
<div
  class="App"
  data-testid="app-component"
>
  <div
    class="container"
  >
    <header
      data-testid="header-component"
      style="background-color: black; color: white; padding-bottom: 10px;"
    >
      <h1>
        Astro Calendar
      </h1>
      <a
        href="/"
        style="color: white;"
      >
        Home
      </a>
       | 
      <a
        href="/About"
        style="color: white;"
      >
        About
      </a>
    </header>
    <button
      data-testid="scrape-button"
    >
      Get Events
    </button>
    <div
      data-testid="events-list"
    />
  </div>
</div>
`;

exports[`App Component renders 'Home' page correctly 1`] = `
<div
  class="App"
  data-testid="app-component"
>
  <div
    class="container"
  >
    <header
      data-testid="header-component"
      style="background-color: black; color: white; padding-bottom: 10px;"
    >
      <h1>
        Astro Calendar
      </h1>
      <a
        href="/"
        style="color: white;"
      >
        Home
      </a>
       | 
      <a
        href="/About"
        style="color: white;"
      >
        About
      </a>
    </header>
    <button
      data-testid="scrape-button"
    >
      Get Events
    </button>
    <div
      data-testid="events-list"
    >
      <div
        data-testid="event-obj-div"
        style="background-color: rgb(219, 22, 22);"
      >
        <h2>
          Cosmic Insanity
        </h2>
        <h3>
          April 1
        </h3>
        <button>
          Create Reminder
        </button>
      </div>
      <div
        data-testid="event-obj-div"
        style="background-color: rgb(244, 244, 244);"
      >
        <h2>
          Don't Look Up
        </h2>
        <h3>
          April 2
        </h3>
        <button>
          Create Reminder
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Component renders header correcty 1`] = `
<header
  data-testid="header-component"
  style="background-color: black; color: white; padding-bottom: 10px;"
>
  <h1>
    Astro Calendar
  </h1>
  <a
    href="/"
    style="color: white;"
  >
    Home
  </a>
   | 
  <a
    href="/About"
    style="color: white;"
  >
    About
  </a>
</header>
`;
